src/mpl/configure.ac

```diff
@@ -959,14 +959,17 @@ if test "X${have_cudart}" = "Xyes" -a \
 fi
 AM_CONDITIONAL([MPL_HAVE_CUDA],[test "X${have_cudart}" = "Xyes" -a "X${have_cuda}" = "Xyes"])
 
-
 PAC_SET_HEADER_LIB_PATH([ze])
 PAC_CHECK_HEADER_LIB([level_zero/ze_api.h],[ze_loader],[zeInit],[have_ze=yes],[have_ze=no])
+# Level Zero will not be enabled if CUDA is enabled firstly.
+if test "X${have_cudart}" != "Xyes" -a \
+    "X${have_cuda}" != "Xyes" ; then
+    have_ze="no"
+fi
```

Comment:
I was more expecting something like an `elif` branch so the exclusiveness is enforced by semantics. The fix here seems like an bandaid -- what if we introduce another GPU API?

Reply:
How about something like this:

```
have_gpu=no
if test $have_gpu = no ; then 
   # test for cuda, set cuda defines and set have_gpu=yes if found
fi

if test "$have_gpu" = "no" ; then
    # test for ze, set ze variables and have_gpu=yes if found
fi
```
?

Reply:
I can change that way, looks equivalent to me.

Reply:
The difference is -- in your version, you are referring to cuda variables inside ze detection block, and referring to ze variables inside cuda detection block. If we remove or add one more gpu api, we have to modify the code in every block.  I would prefer a version that each block is more or less orthogonal.

Reply:
fixed, let me know other comments.